### PR TITLE
Clarify compatibility guarantees around spec.preserveUnknownFields

### LINKED ISSUE #25064: Extend the Kubernetes API with CustomResourceDefinitions
In the "[Field Pruning](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#field-pruning )" section there is a text box referring to the conversion from CRDs defined with apiVersion: apiextensions.k8s.io/v1beta1 to apiVersion: apiextensions.k8s.io/v1.
It says (second and final paragraph in that text box):

_For migrated CustomResourceDefinitions where spec.preserveUnknownFields is set, pruning is not enabled and you can store arbitrary data. For best compatibility, you should update your custom resources to meet an OpenAPI schema, and you should set spec.preserveUnknownFields to true for the CustomResourceDefinition itself._

I think it is wrong. As we aim for CRD objects defined according to apiextensions.k8s.io/v1beta1 to become RIGHT (as much as possible) CRD objects according to apiextensions.k8s.io/v1 (once converted) I think it should say instead (second sentence in that paragraph):

For best compatibility, you should update your custom objects to meet an structural OpenAPI schema, and you should set spec.preserveUnknownFields to false for the CustomResourceDefinition itself where to also specify that structural OpenAPI schema for each defined custom resource version.

### PR DESCRIPTION
**Pull request by apelisse**
I just copy-pasted the [text](https://github.com/kubernetes/website/issues/25064#issuecomment-732269954) as described by Stefan.

Fixes #25064.

### CODE REVIEW

**Comment by apelisse** [comment]
cc @antaloala @sttts @sftim

File: content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md [hunk 1]
```diff
@@ -314,7 +314,7 @@ CustomResourceDefinitions store validated resource data in the cluster's persist
 {{< note >}}
 CRDs converted from `apiextensions.k8s.io/v1beta1` to `apiextensions.k8s.io/v1` might lack structural schemas, and `spec.preserveUnknownFields` might be `true`.
 
-For migrated CustomResourceDefinitions where `spec.preserveUnknownFields` is set, pruning is _not_ enabled and you can store arbitrary data. For best compatibility, you should update your custom resources to meet an OpenAPI schema, and you should set `spec.preserveUnknownFields` to true for the CustomResourceDefinition itself.
+For legacy CustomResourceDefinitions created as `apiextensions.k8s.io/v1beta1` where spec.preserveUnknownFields is set to true, pruning is not enabled and you can store arbitrary data. For compatibility with `apiextensions.k8s.io/v1`, you should update your custom resources definition to (1) have a structural OpenAPI schema, and (2) you should set spec.preserveUnknownFields to false. In Kubernetes 1.22 the `apiextensions.k8s.io/v1beta1` API will be removed, and every CustomResourceDefinition must be created via `apiextensions/v1`, and this enforces both (1) and (2).
```
Comment (kbhawkey) [inline content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md:317]: Hi @apelisse . What do you think about adding two lists? 
Use "custom resource definitions" or "CustomResourceDefinitions"?

For legacy CustomResourceDefinition objects created as `apiextensions.k8s.io/v1beta1` with `spec.preserveUnknownFields` set to `true`, the following is also true:
- Pruning is not enabled.
- You can store arbitrary data.

For compatibility with `apiextensions.k8s.io/v1`, update your custom resource definitions to:
1. Use a structural OpenAPI schema.
1. Set `spec.preserveUnknownFields` to `false`.


I suggest to remove the last sentence about future changes?

Comment (apelisse) [reply content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md:317]: Sounds reasonable to me!

**Comment by apelisse** [comment]
PTAL!

**Review by sftim — COMMENTED** [review]
/approve

Would like a second pair of eyes to add LGTM - @kubernetes/sig-api-machinery-pr-reviews or anyone else who is happy to review, please take a look.

**Pull request merged**